.github/local-actions/branch-manager/lib/main.ts

```diff
@@ -0,0 +1,161 @@
+import * as core from '@actions/core';
+import {PullRequestValidationConfig} from '../../../../ng-dev/pr/common/validation/validation-config.js';
+import {
+  assertValidPullRequestConfig,
+  PullRequestConfig,
+} from '../../../../ng-dev/pr/config/index.js';
+import {loadAndValidatePullRequest} from '../../../../ng-dev/pr/merge/pull-request.js';
+import {AutosquashMergeStrategy} from '../../../../ng-dev/pr/merge/strategies/autosquash-merge.js';
+import {
+  assertValidGithubConfig,
+  getConfig,
+  GithubConfig,
+  setConfig,
+} from '../../../../ng-dev/utils/config.js';
+import {AuthenticatedGitClient} from '../../../../ng-dev/utils/git/authenticated-git-client.js';
+import {
+  ANGULAR_ROBOT,
+  getAuthTokenFor,
+  revokeActiveInstallationToken,
+} from '../../../../github-actions/utils.js';
+import {MergeConflictsFatalError} from '../../../../ng-dev/pr/merge/failures.js';
+import {chdir} from 'process';
+import {spawnSync} from 'child_process';
+
+interface CommmitStatus {
+  state: 'pending' | 'error' | 'failure' | 'success';
+  description: string;
+}
+
+async function main(repo: {owner: string; repo: string}, token: string, pr: number) {
+  if (isNaN(pr)) {
+    core.setFailed('The provided pr value was not a number');
+    return;
+  }
+
+  // Because we want to perform this check in the targetted repository, we first need to check out the repo
+  // and then move to the directory it is cloned into.
+  chdir('/tmp');
+  console.log(
+    spawnSync('git', [
+      'clone',
+      `https://github.com/${repo.owner}/${repo.repo}.git`,
+      './branch-manager-repo',
+    ]).output.toString(),
```

Comment:
We can clone with `--depth=1` for increased speed right? I assume the problem might be that we miss branches, but those could be fetched explicitly. Still much faster than fetching everything

Reply:
We should be fine to do this as the `prepare` step of the `MergeStrategy` explicitly fetches the necessary branches.